packages/wonder-blocks-announcer/src/Announcer.ts

```diff
@@ -0,0 +1,291 @@
+                this.dictionary.set(region.id, {
+                    id: region.id,
+                    levelIndex: parseInt(
+                        region.id.charAt(region.id.length - 1),
+                    ),
+                    level: region.getAttribute("aria-live") as PolitenessLevel,
+                    element: region,
+                });
+            });
+        }
+    }
+
+    /**
+     * Create a wrapper element to group regions for a given level
+     * @param {string} level Politeness level for grouping
+     * @returns {HTMLElement} Wrapper DOM element reference
+     */
+    createRegionWrapper(level: PolitenessLevel) {
+        const wrapper = document.createElement("div");
+        wrapper.id = `wbAWrap-${level}`;
+        return wrapper;
+    }
+
+    /**
+     * Create multiple live regions for a given level
+     * @param {HTMLElement} wrapper Parent DOM element reference to append into
+     * @param {string} level Politeness level for grouping
+     * @returns {HTMLElement[]} Array of region elements
+     */
+    createDuplicateRegions(
+        wrapper: HTMLElement,
+        level: PolitenessLevel,
+    ): HTMLElement[] {
+        const result = new Array(this.regionFactory.count)
+            .fill(0)
+            .map((el, i) => {
+                const region = this.createRegion(level, i);
+                wrapper.appendChild(region);
+                return region;
+            });
+        return result;
+    }
+
+    /**
+     * Create live region element for a given level
+     * @param {string} level Politeness level for grouping
+     * @param {number} index Incrementor for duplicate regions
+     * @param {string} role Role attribute for live regions, defaults to log
+     * @returns {HTMLElement} DOM element reference for live region
+     */
+    createRegion(level: PolitenessLevel, index: number, role = "log") {
+        const region = document.createElement("div");
+        // TODO: test combinations of attrs
+        region.setAttribute("role", role);
+        region.setAttribute("aria-live", level);
+        region.classList.add("wbARegion");
+        const id = `wbARegion-${level}${index}`;
+        region.id = id;
+        region.setAttribute("data-testid", id);
+        this.dictionary.set(id, {
+            id,
+            levelIndex: index,
+            level,
+            element: region,
+        });
+        return region;
+    }
+
+    /**
+     * Announce a live region message for a given level
+     * @param {string} message The message to be announced
+     * @param {string} level Politeness level: should it interrupt?
+     * @param {number} removalDelay How long to wait before removing message
+     * @returns {string} IDREF for targeted element or empty string if it failed
+     */
+    announce(
+        message: string,
+        level: PolitenessLevel,
+        removalDelay?: number,
+    ): string {
+        if (!this.node) {
+            return "";
```

Comment:
If node isn't set at this point, could we trigger the initialization steps so that it can continue to announce the message?

Reply:
Yes, great idea! I renamed the HMR method to "reattachNodes" and called it from this part of the code.